consumer/handler/doc.go

```diff
@@ -0,0 +1,54 @@
+// The handler package defines the mechanisms by which a recieved
+// message can be handled, and how handlers themselves can be managed.
+// The most obvious way to use the handler package is via the consumer
+// package, but this isn't an absolute requirement.  We do however
+// require that the messages handled are felice's message.Message
+// type.
+//
+// The Handler interface defines the signature for all felice
+// Handlers.  There are two common ways to comply with this interface.
+// The first is simply to create a type with the HandleMessage
+// function:
+//
+//    type MyFooHandler struct { }
+//
+//    func (mfh MyFooHandler) HandleMessage(msg *message.Message) error {
+//        fmt.Printf("%+v", *msg)
```

Comment:
:tulip: I think we also need to explain what returning an error implies, about the fact that when a user returns an error, the message is replayed after a delay, indefinitely.
But maybe after the PR with the consumer?

Reply:
> But maybe after the PR with the consumer?

🌷 I think it can be added to this PR.

Reply:
I've documented this on the interface's definition of `HandleMessage`.